Client/app/src/main/java/ru/hse/client/SignUpActivity.kt

```diff
@@ -0,0 +1,338 @@
+
+            val requestForRegisterUser: Request = Request.Builder()
+                .url(URlRegistration)
+                .post(requestBody)
+                .build()
+
+            Log.i("Info", "Request has been sent $URlRegistration")
+            val sometghinWentWrongMessage: String = "Something went wrong, try again"
+            okHttpClient.newCall(requestForRegisterUser).enqueue(object : Callback {
+                override fun onFailure(call: Call, e: IOException) {
+                    Log.e("Error", e.toString() + " " + e.message)
+                    Handler(Looper.getMainLooper()).post {
+                        Toast.makeText(this@SignUpActivity, sometghinWentWrongMessage, Toast.LENGTH_SHORT).show()
+                    }
+                }
+
+                override fun onResponse(call: Call, response: Response) {
+                    Log.i("Info", response.toString())
+                    if (response.code == 200) {
+                        val URlGetUser: String = ("http://" + getString(R.string.IP) + "/users/userByLogin").toHttpUrlOrNull()
+                            ?.newBuilder()
+                            ?.addQueryParameter("login", login)
+                            ?.build().toString()
+
+                        val requestForGetGeneratedUser: Request = Request.Builder()
+                            .url(URlGetUser)
+                            .build()
+
+                        okHttpClient.newCall(requestForGetGeneratedUser).enqueue(object : Callback {
+                            override fun onFailure(call: Call, e: IOException) {
+                                Log.e("Error", e.toString() + " " + e.message)
+                                Handler(Looper.getMainLooper()).post {
+                                    Toast.makeText(this@SignUpActivity, sometghinWentWrongMessage + " with another login", Toast.LENGTH_SHORT).show()
+                                }
+                            }
+
+                            override fun onResponse(call: Call, response: Response) {
+                                Log.i("Info", response.toString())
+                                if (response.code == 200) {
+                                    val responseBody: ByteString? = response.body?.byteString()
+                                    val registredUser: UserModel = UserModel.parseFrom(responseBody?.toByteArray())
+                                    Handler(Looper.getMainLooper()).post {
+                                        Toast.makeText(this@SignUpActivity, "User has been successfully created", Toast.LENGTH_SHORT).show()
+                                    }
+                                    // TODO: create main Activity
+                                }
+                            }
+                        })
+
+                    } else if (response.code == 400) {
```

Comment:
not handling 500 error code and other, i think better use response.ok(), and write message from server